src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -364,6 +375,17 @@ namespace winrt::TerminalApp::implementation
             _searchBox().PasteFromClipboard();
             e.Handled(true);
         }
+        else if (key == VirtualKey::Right && _currentMode == CommandPaletteMode::CommandlineMode)
+        {
+            if (const auto command{ _filteredActionsView().SelectedItem().try_as<winrt::TerminalApp::FilteredCommand>() })
+            {
+                _searchBox().Text(command.Item().Name());
+                _searchBox().Select(_searchBox().Text().size(), 0);
+                _searchBox().Focus(FocusState::Programmatic);
+                _filteredActionsView().SelectedIndex(-1);
+                e.Handled(true);
```

Comment:
Same question as @Rosefield, actually -- this doesn't break left/right in the general case, right? For editing the command once you pop it in the box?

Reply:
I believe the change as is is fine since it only handles the right array key if you have something selected, and it deselects the option after you hit right arrow the first time.